certify-service/src/main/java/io/mosip/certify/services/StatusListUpdateBatchJob.java

```diff
@@ -182,73 +125,53 @@ public void updateStatusList(String statusListId, List<CredentialStatusTransacti
 
             if (optionalStatusList.isEmpty()) {
                 log.error("Status list credential not found: {}", statusListId);
-                throw new CertifyException("STATUS_LIST_NOT_FOUND");
+                throw new CertifyException(ErrorConstants.STATUS_LIST_NOT_FOUND);
             }
 
             StatusListCredential statusListCredential = optionalStatusList.get();
 
-            // Get current status data for this status list
-            Map<Long, Boolean> currentStatuses = getCurrentStatusData(statusListId);
-
             // Apply transaction updates to the status data
-            Map<Long, Boolean> updatedStatuses = applyTransactionUpdates(currentStatuses, transactions);
+            Map<Long, Boolean> updatedStatuses = getUpdatedStatus(transactions);
+
+            JSONObject vcDocument = new JSONObject(statusListCredential.getVcDocument());
 
             // Generate new encoded list
-            String newEncodedList = BitStringStatusListUtils.generateEncodedList(updatedStatuses, statusListCredential.getCapacity());
+            String newEncodedList = BitStringStatusListUtils.updateEncodedList(vcDocument.getJSONObject("credentialSubject").getString("encodedList"),updatedStatuses, statusListCredential.getCapacity());
```

Comment:
Critical: Is this capacity in db is holded as KB ? Asking this because the Util is expecting this in KB